### PR TITLE
Separate Apso in modules

### PR DESCRIPTION
**Pull request by beatriz**
This separates Apso in modules, to give users the option to import smaller dependencies into their projects. 

I followed a rule of thumb of creating modules for each major top-level package, but feel free to suggest alternatives or changes 🙂 

I've kept publishing of the full `apso` to avoid unnecessary breaking changes and keep the choice to whoever is using.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Apso [![Build Status](https://travis-ci.org/velocidi/apso.svg?branch=master)](https://travis-ci.org/velocidi/apso) [![Maven Central](https://img.shields.io/maven-central/v/com.velocidi/apso_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/com.velocidi/apso_2.12)
 
-Apso is Velocidi's Scala utilities library. It provides a series of useful methods.
+Apso is Velocidi's Scala project of utilities libraries. It provides a series of useful methods.
```
Comment (JD557) [inline README.md]: I would probably rewrite this as:
```suggestion
Apso is Velocidi's collection of Scala utility libraries. It provides a series of useful methods.
```

(Feel free to ignore this comment)

File: build.sbt [hunk 2]
```diff
@@ -1,76 +1,144 @@
...
 
 scalaVersion in ThisBuild := "2.12.10"
 
+def module(project: Project, moduleName: String, dependencies: sbt.librarymanagement.ModuleID*) =
+  project.in(file(moduleName))
+  .settings(commonSettings: _*)
+  .settings(name := s"apso-$moduleName",
+    libraryDependencies ++= dependencies)
+
 lazy val core = project.in(file("core"))
```
Comment (JD557) [inline build.sbt]: Can't the `module` helper be used here?

File: build.sbt [hunk 3]
```diff
@@ -1,76 +1,144 @@
...
+      "com.typesafe.akka"                       %% "akka-stream-testkit"            % defaultVersion % "test",
+      "junit"                                    % "junit"                          % "4.12"         % "test",
+      "net.ruippeixotog"                        %% "akka-testkit-specs2"            % defaultVersion % "test",
+      "org.apache.logging.log4j"                 % "log4j-core"                     % "2.11.2"       % "test",
+      "org.scalacheck"                          %% "scalacheck"                     % defaultVersion % "test",
+      "org.specs2"                              %% "specs2-core"                    % defaultVersion % "test",
+      "org.specs2"                              %% "specs2-scalacheck"              % defaultVersion % "test",
+      "org.specs2"                              %% "specs2-junit"                   % defaultVersion % "test"))
 
 lazy val testkit = project.in(file("testkit"))
```
Comment (JD557) [inline build.sbt]: Can't the `module` helper be used here?

File: README.md [hunk 4]
```diff
@@ -752,7 +798,13 @@ The `SimpleJmx` trait allows mixing in a simple JMX server. The JMX server is co
 
 ## Spray
```
Comment (JD557) [inline README.md]: I'm a bit confused about c8adfcd3c0b962b58ca001a6daf23cc4e9f06e81

Spray has been superseded by akka-http, and that module only provides akka-http directives, so I think that we should keep the akka-http name and update the README instead.

File: build.sbt (hunk 3 above)
Comment (beatriz) [reply build.sbt]: Yes, changed on 815fc9f6e11c0251540c4c31a34cbed6f94ebd2b.

File: build.sbt (hunk 2 above)
Comment (beatriz) [reply build.sbt]: Yes, done on 815fc9f6e11c0251540c4c31a34cbed6f94ebd2b.

File: README.md (hunk 4 above)
Comment (beatriz) [reply README.md]: You're right. I had started with `akka-http` and when got to the README panicked and changed to `spray` 😅 Changed back the module name to `akka-http` and updated the README on 377f730fcc3c61d43f8840ce0911d97d6aa4a619.

**Pull request merged**